### DISCUSSION TITLE
improperly re-located library tracks

### DISCUSSION THREAD

**Discussion by jmgant** [beetbox/beets#4357] [General]
I use the following path format:

`%the{$albumartist}/ %if{$disctitle,$original_year - $album/[Disc]  $disc - $disctitle/$track - $title,$original_year - $album/$track - $title}`

this checks box set releases for the presence of disc subtitles, then stores tracks in those separate, named sub-directories when importing box sets.

this works 100% of the time when importing new tracks. however, if I use beets to move an existing library item to a new location, the tracks don't always end up where they belong. this happens when using beet convert as well.

when this error happens, the album structure stays 'mostly' in tack. however a small portion of the music files of a box set end up in the upper-level 'album' directory instead of doing:
`/Music/Artist/Album/Disctitle01/01-track.mp3
/Music/Artist/Album/Disctitle02/02-track.mp3`
some of the tracks land in
`/Music/Artist/Album/01-track.mp3, 02-track.mp3`

beets is ignoring the $disctitle field on tracks that do possess it. this only happens to a small percentage of the tracks. mostly, the directory structure stays in tact after the move/transcode.

I am wondering if I can prevent this error on future runs.

**Comment by sampsyo** [comment]
> however, if I use beets to move an existing library item to a new location, the tracks don't always end up where they belong.

Hmm; can you clarify exactly what you're doing to move these files? Are you using `beet modify`?

I also want to point out that you might find it a little easier to use query-specific path formats instead of `%if`, as described on [this docs page](https://beets.readthedocs.io/en/stable/reference/config.html#path-format-config). Something like this:

```yaml
paths:
    disctitle::.: %the{$albumartist}/$original_year - $album/[Disc] $disc - $disctitle/$track - $title
    default: %the{$albumartist}/$original_year - $album/$track - $title
```

**Reply by jmgant** [reply]
it happens in two situations.

1). when I set a new library path in the config and run `beet import` against existing files with `move` instead of `copy`.  doing that will result in misplaced tracks. that's a usecase I don't see myself doing again, but it has occurred.

2). when transcoding existing library entries, 'beet convert -a album_title' results in misplaced box set tracks in the target transcode directory. this is a problem since I need to transcode 656 albums.

if I'm not mistaken, wouldn't the path you described create incorrect entries in the case of (non box set) albums that had no `$disctitle`? I thought the point of `%if ` was to check for the existence of a field before applying a value? I could be wrong though. I'll try.

no matter how my fresh target files are organized, they never have misplaced tracks after a first import (copy). this unwanted behavior only happens when targeting exiting library paths which are already sorted as
`...$album/[Disc] $disc - $disctitle/$track - $title`
